File: dotnet/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/InsertTextCommand.cs
namespace CommandDesignPattern.Example1;

/// <summary>
/// Concrete Command - Insert text command
/// </summary>
public class InsertTextCommand : ICommand
{
    private readonly TextEditor _editor;
    private readonly string _text;

    public InsertTextCommand(TextEditor editor, string text)
    {
        _editor = editor;
        _text = text;
    }

    public void Execute()
    {
        _editor.AppendText(_text);
    }

    public void Unexecute()
    {
        _editor.DeleteText(_text.Length);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/src/InsertTextCommand.ts
import { ICommand } from "./ICommand";
import { TextEditor } from "./TextEditor";

/**
 * Concrete Command — inserts text into the editor.
 */
export class InsertTextCommand implements ICommand {
  constructor(
    private readonly _editor: TextEditor,
    private readonly _text: string
  ) {}

  execute(): void {
    this._editor.appendText(this._text);
  }

  unexecute(): void {
    this._editor.deleteText(this._text.length);
  }
}